OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -181,6 +188,13 @@ public virtual WVec GetRepulsionForce()
 			if (!Info.Repulsable)
 				return WVec.Zero;
 
+			if (reservation != null)
+			{
+				var distanceFromReservationActor = (reservedActor.CenterPosition - self.CenterPosition).HorizontalLength;
+				if (distanceFromReservationActor < Info.WaitDistanceFromResupplyBase.Length)
+					return WVec.Zero;
+			}
```

Comment:
I'm not sure if I like this (mind you, I haven't playtested it yet, so it might be alright). My first gut instinct would be to separate waiting helicopters vertically instead of disabling repulsion. Like a holding pattern, kinda.